app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt

```diff
@@ -135,47 +136,73 @@ class ChatRoomPresenter @Inject constructor(
         draftKey = "${currentServer}_${LocalRepository.DRAFT_KEY}$roomId"
         chatRoomId = roomId
         chatRoomType = roomType
+        chatRoomName = roomName
+        chatRoles = emptyList()
+        var canModerate = isOwnerOrMod()
+
         GlobalScope.launch(Dispatchers.IO + strategy.jobs) {
-            try {
-                chatRoles = if (roomTypeOf(roomType) !is RoomType.DirectMessage) {
-                    client.chatRoomRoles(roomType = roomTypeOf(roomType), roomName = roomName)
-                } else {
-                    emptyList()
-                }
-            } catch (ex: Exception) {
-                Timber.e(ex)
-                chatRoles = emptyList()
-            } finally {
-                // User has at least an 'owner' or 'moderator' role.
-                val canModerate = isOwnerOrMod()
-                // Can post anyway if has the 'post-readonly' permission on server.
-                val room = dbManager.getRoom(roomId)
-                room?.let {
-                    chatIsBroadcast = it.chatRoom.broadcast ?: false
-                    val roomUiModel = roomMapper.map(it, true)
-                    launchUI(strategy) {
-                        view.onRoomUpdated(roomUiModel = roomUiModel.copy(
+            // Can post anyway if has the 'post-readonly' permission on server.
+            val room = dbManager.getRoom(roomId)
+            room?.let {
+                chatIsBroadcast = it.chatRoom.broadcast ?: false
```

Comment:
`isSomething` generally fits better.

So:

`isChatBroadcast` or `isBroadcastChannel` instead of  `chatIsBroadcast`.